SYCL/ESIMD/regression/bit_shift_compilation_test.cpp

```diff
@@ -0,0 +1,195 @@
+  shared_vector<TRes> lsr_output(THREAD_NUM * SIMD, allocator0);
+  shared_vector<TRes> asr_output(THREAD_NUM * SIMD, allocator0);
+
+  shared_vector<TRes> scalar_shl_output(1, allocator0);
+  shared_vector<TRes> scalar_shr_output(1, allocator0);
+  shared_vector<TRes> scalar_rol_output(1, allocator0);
+  shared_vector<TRes> scalar_ror_output(1, allocator0);
+  shared_vector<TRes> scalar_lsr_output(1, allocator0);
+  shared_vector<TRes> scalar_asr_output(1, allocator0);
+
+  nd_range<1> Range((range<1>(THREAD_NUM)), (range<1>(SIMD)));
+  auto e = q.submit([&](handler &cgh) {
+    TArg1 *in_ptr = input.data();
+    TRes *shl_ptr = shl_output.data();
+    TRes *shr_ptr = shr_output.data();
+    TRes *rol_ptr = rol_output.data();
+    TRes *ror_ptr = ror_output.data();
+    TRes *lsr_ptr = lsr_output.data();
+    TRes *asr_ptr = asr_output.data();
+
+    TRes *scalar_shl_ptr = scalar_shl_output.data();
+    TRes *scalar_shr_ptr = scalar_shr_output.data();
+    TRes *scalar_rol_ptr = scalar_rol_output.data();
+    TRes *scalar_ror_ptr = scalar_ror_output.data();
+    TRes *scalar_lsr_ptr = scalar_lsr_output.data();
+    TRes *scalar_asr_ptr = scalar_asr_output.data();
+
+    cgh.parallel_for(Range, [=](nd_item<1> it) SYCL_ESIMD_FUNCTION {
+      Sat sat;
+
+      __ESIMD_NS::simd<TArg1, SIMD> input_vec;
+      input_vec.copy_from(in_ptr + it.get_global_id(0) * SIMD);
+
+      __ESIMD_NS::simd<TRes, 1> scalar_result;
+      __ESIMD_NS::simd<TRes, SIMD> result;
+
+      result = shl<TArg1>(input_vec, arg2, sat);
+      scalar_result = shl<TArg1>(arg1, arg2, sat);
+      result.copy_to(shl_ptr + it.get_global_id(0) * SIMD);
+      scalar_result.copy_to(scalar_shl_ptr);
+
+      result = shr<TArg1>(input_vec, arg2, sat);
+      scalar_result = shr<TArg1>(arg1, arg2, sat);
+      result.copy_to(shr_ptr + it.get_global_id(0) * SIMD);
+      scalar_result.copy_to(scalar_shr_ptr);
+
+      result = rol<TArg1>(input_vec, arg2);
+      scalar_result = rol<TArg1>(arg1, arg2);
+      result.copy_to(rol_ptr + it.get_global_id(0) * SIMD);
+      scalar_result.copy_to(scalar_rol_ptr);
+
+      result = ror<TArg1>(input_vec, arg2);
+      scalar_result = ror<TArg1>(arg1, arg2);
+      result.copy_to(ror_ptr + it.get_global_id(0) * SIMD);
+      scalar_result.copy_to(scalar_ror_ptr);
+
+      result = lsr<TArg1>(input_vec, arg2, sat);
+      scalar_result = lsr<TArg1>(arg1, arg2, sat);
+      result.copy_to(lsr_ptr + it.get_global_id(0) * SIMD);
+      scalar_result.copy_to(scalar_lsr_ptr);
+
+      result = asr<TArg1>(input_vec, arg2, sat);
+      scalar_result = asr<TArg1>(arg1, arg2, sat);
+      result.copy_to(asr_ptr + it.get_global_id(0) * SIMD);
+      scalar_result.copy_to(scalar_asr_ptr);
+    });
+  });
+  e.wait();
+
+  for (int i = 0; i < THREAD_NUM * SIMD; ++i) {
+    if (shl_output[i] ^ expected_shl || scalar_shl_output[0] ^ expected_shl) {
+      std::cout << "Test failed for shl test " input[i] << "." << std::endl;

```

Comment:
The testing failed. Please fix the syntax error here.

Reply:
I see the fix for compilation error. Thank you.
Well, the test still fails, e.g. 
"error: LLVM ERROR: VISA builder API call failed: CisaBuilder->Compile("genxir", &ss, EmitVisa)"